core/src/OpenACC/Kokkos_OpenACC_ParallelScan_Range.hpp

```diff
@@ -0,0 +1,183 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENACC_PARALLEL_SCAN_RANGE_HPP
+#define KOKKOS_OPENACC_PARALLEL_SCAN_RANGE_HPP
+
+#include <OpenACC/Kokkos_OpenACC.hpp>
+#include <OpenACC/Kokkos_OpenACC_FunctorAdapter.hpp>
+#include <Kokkos_Parallel.hpp>
+
+namespace Kokkos::Experimental::Impl {
+template <class IndexType, class Functor, class ValueType>
+void OpenACCParallelScanRangePolicy(IndexType begin, IndexType end,
+                                    IndexType chunk_size, Functor afunctor,
+                                    ValueType init_value, int async_arg) {
+  auto const functor(afunctor);
+  const IndexType N        = end - begin;
+  const IndexType n_chunks = (N + chunk_size - 1) / chunk_size;
+  Kokkos::View<ValueType*, Kokkos::Experimental::OpenACC> chunk_values(
+      "chunk_values", n_chunks);
+  Kokkos::View<ValueType*, Kokkos::Experimental::OpenACC> offset_values(
+      "offset_values", n_chunks);
+  ValueType* element_values = new ValueType[2 * chunk_size];
```

Comment:
Thank you for catching the missing `delete` call; added.
(We cannot use `std::unique_ptr` here, since OpenACC compilers cannot handle it due to a deep copy issue.)

Reply:
```C++
std::unique_ptr<ValueType[]> element_values_owner(new ValueType[2 * chunk_size]);
ValueType* element_values = element_values_owner.get();

Reply:
> ```c++
> std::unique_ptr<ValueType[]> element_values_owner(new ValueType[2 * chunk_size]);
> ValueType* element_values = element_values_owner.get();
> ```

The above approach will work; I will update as suggested.

Reply:
Updated.